### ISSUE TITLE
Can pnp.sp.web.addClientSidePage only create classic team sites?

### ISSUE THREAD

**Issue by PeopleNetCoder** [SharePoint/PnP-JS-Core#770]
#### Category

[ ] Enhancement

[X] Bug

[X] Question

#### Version

Please specify what version of the library you are using: [ sp-pnp-js@3.0.6 ]

#### Expected / Desired Behavior / Question

I would like to create a modern page in a Communication Site / Modern Team Site using the sp-pnp-js library. I am following this wiki, written by Patrick Rodgers: https://github.com/SharePoint/PnP-JS-Core/wiki/Working-With:-Client-side-Pages

Specifically I am trying to use the function **pnp.sp.web.addClientSidePage(`MyFirstPage.aspx`);** inside my SPFx webpart on a Modern Site workbench: https://mytenant/sites/CommunicationSite/_layouts/15/workbench.aspx.

In my Web Part I have a simple hello world button, which executes the function onClick.
I expect to create a new modern page in the Site Pages library, when the button is clicked.

#### Observed Behavior

When I click the button I get the following console errors:
1.) GET https://mytenant/sites/CommunicationSite/_api/web/lists/getByTitle('Site%20Pages')/rootFolder/files?$select=Name&$filter=Name%20eq%20%27MyFirstPage.aspx%27 404 (Not Found) in fetchclient.js:10 

2.) Uncaught (in promise) Error: Error making HttpClient request in queryable: [404] Not Found
at new ProcessHttpClientResponseException (exceptions.js:24)
at core.js:33

It seems like the function cannot find the library Site Pages on a Modern Site / Communication Site.
If I add my SPFx Web Part on a classic Team Site workbench, it works fine and creates a new page in the Site Pages library.

I have tested the rest of the functions from Patrick Rodgers' article, like adding sections or Web Parts, and they all work perfectly fine on Modern Pages, so it is only the addClientSidePage() I cannot get to work.

My questions is: Does the addClientSidePage() only work on classic Team Sites? 
Or did I miss something obvious :)

#### Steps to Reproduce

1.) Create a SPFx project, using the Yeoman template yo@microsoft/sharepoint.
2.) Add a webpart with React.
3.) Open the project.
4.) Install (npm i sp-pnp-js) and import sp-pnp-js in the .tsx file.

5.) Replace render() with:
public render(): React.ReactElement<ICreateNewsLinkProps> {
    return <div className="panel">
        <div className="panel-body">
            <button id="createModernPage" type="submit" onClick={this.createModernPage} >CREATE MODERN PAGE</button>
        </div>
    </div>;
  }

6.) Add async function:
 private async createModernPage() {
    const page = await pnp.sp.web.addClientSidePage(`MyFirstPage.aspx`);
  }

7.) Run gulp serve in console
8.) Open workbench on your Communcation Site / Modern Team Site.
9.) Click the button.

Thanks in advance and thank you for the great work you do.

Best Regards,
Mads

**Comment by koltyakov** [comment]
In both Team and Communication sites templates, there is `Site Pages` library. And the method works for me.
Thought it can be so that it has different display name (hello localization and get by title methods).
We might change the method to use `web.getList([uri])` approach, shouldn't we?

**Comment by koltyakov** [comment]
@PeopleNetCoder,

Just looked at the code, this situation was thought out really.

```typescript
public addClientSidePage(pageName: string, title = pageName.replace(/\.[^/.]+$/, ""), libraryTitle = "Site Pages"): Promise<ClientSidePage> {
  return ClientSidePage.create(this.lists.getByTitle(libraryTitle), pageName, title);
}
```

There is additional parameter where a library display name can be provided strictly. Which is by default is `Site Pages` but can be redefined with any valid for the site name.

```typescript
pnp.sp.web.addClientSidePage(`MyFirstPage.aspx`, `My first page`, `My Custom Site Pages`);
```

**Comment by PeopleNetCoder** [comment]
@koltyakov thank you for the fast reply.

Yes I can see that localization could be a minor issue, but as you point out the API supports custom libraries with the optional **librarytitle** parameter.

My tenant locale is how ever English and I do have the 'Site Pages' library present. 

I tried the following examples with the optional parameters on:
pnp.sp.web.addClientSidePage(`MyFirstPage.aspx`, `My first page`, `Site Pages`);
pnp.sp.web.addClientSidePage(`MyFirstPage.aspx`, `My first page`, `SitePages`);
pnp.sp.web.addClientSidePage(`MyFirstPage.aspx`, `My first page`, `Site%20Pages`);
pnp.sp.web.addClientSidePage(`MyFirstPage.aspx`, `My first page`, `sitepages`);

All result in the same error as I first described.

I can how ever do the following:
1.) Create a document library called **mynewpageslib**
2.) Replace the default **Document** Content Type with the **Site Page** Content Type
3.) Run pnp.sp.web.addClientSidePage(`MyFirstPage.aspx`, `My first page`, `mynewpageslib`);
This works! That's progress at least :)

The supported system has 12.000 employees combined in 9 intranets, so this might be a hard architectural change to implement.

**Comment by koltyakov** [comment]
Oh, that's weird.
Could you try opening `https://[mytenant]/sites/CommunicationSite/_api/web/lists/getByTitle('Site%20Pages')` right in browser  will it return some API responce or 404 error?

**Comment by PeopleNetCoder** [comment]
Yes it does indeed return a 404 error.
"List 'Site Pages' does not exist at site with URL 'https://tenant/sites/CommunicationSite'".

It is just a standard Communication Site, created from the new SharePoint front page (/_layouts/sharepoint.aspx). 
That is why my original question was if the feature was only supported in classic.

I get the same error in the Modern Team Site. It might be Tenant related then?

**Comment by koltyakov** [comment]
Before answering this issue I checked on a newly created Communication and Team site (/_layouts/sharepoint.aspx - yeah, from here, manual creation), on the first-release tenant. Very weird how the API can't see it? O_o

Will Site Pages be available with this request:

```typescript
pnp.sp.web.lists.filter(`Title eq 'Site Pages'`).get().then(console.log);
```

**Comment by PeopleNetCoder** [comment]
I got it figured out now - it was indeed a local error.
Apparently our Tenant is created in danish back in 2013. 
Later on I have changed my UI language to English.

So even though I see the Title as Site Pages, the actual Title is "Webstedssider" (Danish).
I never noticed this because I see the ListID when inside the library settings, and the Internal Name "SitePages" when hovering the list in Site Contents.

I will close this Thread now.
Thank you for the sparing and pinpointing of the problem :)

Best Regards,
Mads
